pkg/cmd/experimental/syncgroups/grouplocator/grouplocator.go

```diff
@@ -0,0 +1,70 @@
+package grouplocator
+
+import (
+	"github.com/openshift/origin/pkg/auth/ldaputil"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+)
+
+func NewGroupBasedLocator(groupGetter interfaces.LDAPGroupGetter) interfaces.LDAPGroupDetector {
+	return &GroupBasedLocator{groupGetter: groupGetter}
+}
+
+type GroupBasedLocator struct {
+	groupGetter interfaces.LDAPGroupGetter
+}
+
+func (l *GroupBasedLocator) Exists(ldapGroupUID string) (bool, error) {
+	_, err := l.groupGetter.GroupEntryFor(ldapGroupUID)
+	if ldaputil.IsQueryOutOfBoundsError(err) || ldaputil.IsEntryNotFoundError(err) || ldaputil.IsNoSuchObjectError(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func NewMemberBasedLocator(memberExtractor interfaces.LDAPMemberExtractor) interfaces.LDAPGroupDetector {
+	return &MemberBasedLocator{memberExtractor: memberExtractor}
+}
+
+type MemberBasedLocator struct {
+	memberExtractor interfaces.LDAPMemberExtractor
+}
+
+func (l *MemberBasedLocator) Exists(ldapGrouUID string) (bool, error) {
+	members, err := l.memberExtractor.ExtractMembers(ldapGrouUID)
+	if ldaputil.IsQueryOutOfBoundsError(err) || ldaputil.IsEntryNotFoundError(err) || ldaputil.IsNoSuchObjectError(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+
+	if len(members) == 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func NewCompoundLocator(locators ...interfaces.LDAPGroupDetector) interfaces.LDAPGroupDetector {
```

Comment:
Still needs godoc.  is the intent here: must satisfy all, must satisfy one, fails on first error, ignores errors unless they all fail, ???  There are a lot of options.  I think this is: all are checked, any errors cause a fail, any exists returns true.